Source/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -740,13 +750,13 @@ - (void)setBackgroundColor:(UIColor *)newBackgroundColor
   if (shouldApply) {
     CGColorRef oldBackgroundCGColor = _layer.backgroundColor;
     
-    BOOL specialPropertiesHandling = ASDisplayNodeNeedsSpecialPropertiesHandling(checkFlag(Synchronous), _flags.layerBacked);
-    if (specialPropertiesHandling) {
-        _view.backgroundColor = newBackgroundColor;
+    if (_flags.layerBacked) {
+      _layer.backgroundColor = newBackgroundCGColor;
     } else {
-        _layer.backgroundColor = newBackgroundCGColor;
+      _view.backgroundColor = newBackgroundColor;
+      _layer.backgroundColor = _view.backgroundColor.CGColor;
```

Comment:
Let's add a comment here on why we're getting the CGColor from the view instead of just using `newBackgroundCGColor`.